library/network_connections.py

```diff
@@ -1136,7 +1136,10 @@ def connection_create(self, connections, idx, connection_current=None):
                     NM.SETTING_802_1X_DOMAIN_SUFFIX_MATCH,
                     connection["ieee802_1x"]["domain_suffix_match"],
                 )
-
+        if connection["match"]:
+            if connection["match"]["path"] is not None:
+                s_match = self.connection_ensure_setting(con, NM.SettingMatch)
+                s_match.set_property(NM.SETTING_MATCH_PATH, connection["match"]["path"])
```

Comment:
What happens/should happen when path is `[""]` (list with empty string). Should it really set this property and will it work as expected or should this remove the property? If it does not remove the property, do we need a way to remove it?

Reply:
In NM, It seems that `[""]` (list with empty string) is not allowed for `path`, https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/blob/main/src/libnm-core-impl/nm-setting-match.c#L528
```
"fatal error: created connection failed to normalize: nm-connection-error-quark: match.path: is empty (7)"
```

Reply:
when `match.path = [""]`, the role will reject the setting in Argvalidator.
when `match.path = ["&"]`, the role will allow the setting, but NM will reject the setting.

Check https://github.com/linux-system-roles/network/pull/363/commits/8fb4516b96348dfb26c0c515dd2279a9f51ecd74#diff-b06f1e1bf991f123e64cf9dde6dcf9574bb279a720609a570e8af27cc2d9a359R3578